### ISSUE TITLE
Error creating dashboard from json

### ISSUE THREAD

**Issue by matt-whyard** [newrelic/terraform-provider-newrelic#2376]
### NewRelic Version
3.23.0

### Terraform Version
1.2.8

### Affected Resource(s)
newrelic_one_dashboard_json

### Terraform Configuration
```
provider "newrelic" {
  account_id = "123456"
  api_key    = "XXX"
  region     = "EU"
}

resource "newrelic_one_dashboard_json" "environment" {
    json = file("${path.module}/dashboards/environment.json")
}
```
Dashboard json below:

```
{
    "name": "XXX",
    "description": null,
    "permissions": "PUBLIC_READ_WRITE",
    "pages": [
      {
        "name": "XXX",
        "description": null,
        "widgets": [
          {
            "title": "Load Balancer Latency",
            "layout": {
              "column": 1,
              "row": 1,
              "width": 4,
              "height": 3
            },
            "linkedEntityGuids": null,
            "visualization": {
              "id": "viz.line"
            },
            "rawConfiguration": {
              "facet": {
                "showOtherSeries": false
              },
              "legend": {
                "enabled": true
              },
              "nrqlQueries": [
                {
                  "accountIds": [
                    3875837
                  ],
                  "query": "SELECT average(`aws.elb.Latency`), average(`aws.elb.Latency.p99`), average(`aws.elb.Latency.p95`) FROM Metric WHERE `aws.elb.LoadBalancerName` = 'XXX' AND `collector.name` = 'cloudwatch-metric-streams' AND `aws.accountId` = 'XXX' TIMESERIES"
                }
              ],
              "platformOptions": {
                "ignoreTimeRange": false
              },
              "units": {
                "unit": "SECONDS"
              },
              "yAxisLeft": {
                "zero": true
              }
            }
          }
        ]
      },
      {
        "name": "vault",
        "description": null,
        "widgets": []
      },
      {
        "name": "xenolith",
        "description": null,
        "widgets": []
      }
    ],
    "variables": []
  }
```
### Actual Behavior
```
 Error: Argument "dashboard" has invalid value $dashboard.
│ In field "pages": Expected type "[DashboardPageInput!]!", found [{name: "krakatoa", widgets: [{configuration: {}, layout: {column: 1, height: 3, row: 1, width: 4}, linkedEntityGuids: null, rawConfiguration: {facet: {showOtherSeries: false}, legend: {enabled: true}, nrqlQueries: [{accountIds: [3875837], query: "SELECT average(`aws.elb.Latency`), average(`aws.elb.Latency.p99`), average(`aws.elb.Latency.p95`) FROM Metric WHERE `aws.elb.LoadBalancerName` = 'staging-krakatoa-elb' AND `collector.name` = 'cloudwatch-metric-streams' AND `aws.accountId` = '281255908620' TIMESERIES"}], platformOptions: {ignoreTimeRange: false}, units: {unit: "SECONDS"}, yAxisLeft: {zero: true}}, title: "Load Balancer Latency", visualization: {id: "viz.line"}}]}, {name: "vault"}, {name: "xenolith"}].
│ In field "facet": Unknown field.
│ In field "legend": Unknown field.
│ In field "nrqlQueries": Unknown field.
│ In field "platformOptions": Unknown field.
│ In field "units": Unknown field.
│ In field "yAxisLeft": Unknown field.
│ In element #2: Expected type "DashboardPageInput", found {name: "vault"}.
│ In field "widgets": Expected type "[DashboardWidgetInput!]!", found null.
│ In element #3: Expected type "DashboardPageInput", found {name: "xenolith"}.
│ In field "widgets": Expected type "[DashboardWidgetInput!]!", found null.
│ 
│   with module.aws_link.newrelic_one_dashboard_json.environment,
│   on modules/aws_link/dashboards.tf line 1, in resource "newrelic_one_dashboard_json" "environment":
│    1: resource "newrelic_one_dashboard_json" "environment" {
```

### Expected Behavior
Dashboard should be created successfully. I created it manually in NewRelic then copied the json, so should be valid.

### Steps to Reproduce
terraform apply

**Comment by matt-whyard** [comment]
On further investigation, it seems it doesn't like empty pages defined. If I remove the empty pages it works ok. Will leave this here as its possible to have empty pages in the NR GUI but not via the terraform provider.

**Comment by Nandu-pns** [comment]
Hi @matt-whyard, Thanks for reporting and further investigation, we'll prioritise based on current workload and look into this fairly soon.

**Comment by pranav-new-relic** [comment]
Hi @matt-whyard - a fix of this issue is out with the latest version of the provider, [v3.25.0](https://github.com/newrelic/terraform-provider-newrelic/releases/tag/v3.25.0). 

With the latest provider, I've just tried creating a dashboard with no widgets listed in the page (an empty page) via the `newrelic_one_dashboard_json` resource (with the configuration you've specified), and it works as expected. Closing this issue now, as the fix is out. Thank you for reporting this.
